packages/core/src/new-api/internal/execution/chain-dispatcher.ts

```diff
@@ -168,7 +168,11 @@ export class ChainDispatcherImpl implements ChainDispatcher {
 
     const contractInstance = new Contract(contractAddress, abi, signer);
 
-    const result = await contractInstance[functionName](...args, {
+    const validFunctionName = functionName.endsWith("()")
+      ? functionName
+      : `${functionName}()`;
```

Comment:
Can you walk me through this? What case pushed that change.

Reply:
Soooo while finding the root of the original bug, I learned that this code didn't actually work before this specific change. The short answer is it's a combination of the way we're invoking the function on the contract object and the way ethers has their methods available on the contract object.

The contract members look like this:
```typescript
const contract = {
  "functionName()": functionName(...),
  functionName: functionName(...)
}
```
so because we're invoking the function through a dynamic string index (`contract[functionName](...)`), for Javascript Reasons :tm:, it only works if it has the parenthesis at the end.

Reply:
The reason it didn't fail before is because ethers doesn't throw an error if you try to index the function without the parenthesis, it just.. returns the contract object.. for reasons I don't understand.

Reply:
Do we support the "setAddress(bytes,bool)" format for function invocation?

Reply:
That is an excellent question, let me check that.

Reply:
This code does not support that, which is a problem. Also worth noting that all of this only affects static calls and not regular calls, but still, we do need to support that case as well.

Reply:
We decided to add a temporary fix for supporting static calls with parameters, with future plans to revisit this with a more comprehensive solution, [captured here](https://github.com/NomicFoundation/ignition/issues/372)